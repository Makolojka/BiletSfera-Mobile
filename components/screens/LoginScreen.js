import React, {useState} from 'react';
import {
    Text,
    View,
    StyleSheet,
    TextInput,
    Image,
    TouchableOpacity,
    ScrollView, ImageBackground,
} from 'react-native';
import {COLORS} from "./Colors";
import FlashMessage, {showMessage} from 'react-native-flash-message';
import AuthService from "../../services/AuthService";
import AsyncStorage from "@react-native-async-storage/async-storage";


const LoginScreen = (props) => {
    const [login, setLogin] = useState('');
    const [password, setPassword] = useState('');

    React.useEffect(() => {
        const fetchData = async () => {
            if(await AuthService.isLoggedIn()){
                // console.log("token: ", AuthService.isLoggedIn())
                props.navigation.navigate('UserNav');
            }
        };
        let a = fetchData();

        // The cleanup function
        return () => {};

    }, [props.navigation]);

    const logIn = async () => {
        const isAuthenticated = await AuthService.authenticate({ login, password });
        if (isAuthenticated === 'Unauthorized') {
            showMessage({
                message: 'Nieprawidłowe dane logowania.',
                type: 'danger',
                backgroundColor: COLORS.red,
                color: COLORS.main,
                style: { display: 'flex', alignContent: 'center', alignItems: 'center' }
            });
        } else if (isAuthenticated) {
            showMessage({
                message: 'Zalogowano.',
                type: 'success',
                backgroundColor: COLORS.green,
                color: COLORS.main,
                style: { display: 'flex', alignContent: 'center', alignItems: 'center' }
            });
            await goToRegister();
        } else {
            showMessage({
                message: 'Nieznany błąd.',
                type: 'danger',
                backgroundColor: COLORS.red,
                color: COLORS.main,
                style: { display: 'flex', alignContent: 'center', alignItems: 'center' }
            });
        }
    };

    const goToRegister = async () => {
        props.navigation.navigate('UserNav');
        setLogin('');
        setPassword('');
    }

    const validateFields = async () => {
        if (login === '' || password === '' || login == null || password == null) {
            showMessage({
                message: 'Uzupełnij wszystkie pola.',
                type: 'warning',
                backgroundColor: COLORS.red,
                color: COLORS.main,
                style: {display: "flex", alignContent: "center", alignItems: "center"}
            });
        } else {
            try {
                await logIn();
            } catch (error) {
                console.error(error);
            }
        }
    }

    return (
        <ImageBackground
            source={require('../screens/img/bg_1.jpg')}
            style={styles.background}
            resizeMode="cover"
        >
            <ScrollView style={styles.container}>

            <View style={styles.section}>
                <View style={styles.logoContainer}>
                    <View style={styles.column}>
                        <Image
                            style={styles.logo}
                            source={require('../screens/img/logoWhite.png')}
                        />
                        <View style={styles.column}>
                            <Text style={styles.textWelcome}>Witaj w BiletSferze</Text>
                            <Text style={styles.textWelcome2}>
                                Zaloguj się, aby kontynuować.
                            </Text>
                        </View>
                    </View>
                </View>
                <View style={styles.box}>
                    <View>
                        <Text style={styles.textInput}>Login</Text>
                        <TextInput
                            style={styles.input}
                            onChangeText={setLogin}
                            value={login}
                            // placeholder="Login"
                            keyboardType="default"
                        />
                        <Text style={styles.textInput}>Hasło</Text>
                        <TextInput
                            style={styles.input}
                            onChangeText={setPassword}
                            value={password}
                            // placeholder="Hasło"
                            keyboardType="default"
                            secureTextEntry={true}
                        />
                        <TouchableOpacity style={styles.button} onPress={validateFields}>
                            <Text style={styles.text}>Zaloguj się</Text>
                        </TouchableOpacity>
                        <TouchableOpacity>
                            <Text style={styles.register} onPress={goToRegister}>
                                Nie masz konta? Zarejestruj się.
                            </Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
        </ScrollView>
            <FlashMessage position="top" />
        </ImageBackground>
    );
}

const styles = StyleSheet.create({
    background: {
        flex: 1,
        width: '100%',
        height: '100%',
    },
    container: {
        flex: 1,
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
    },
    section: {
        justifyContent: 'center',
        display: 'flex',
        flexDirection: 'column',
        flexWrap: 'wrap',
        alignItems: 'center',
        alignContent: 'center',
        backgroundColor: 'transparent',
        padding: 10,
        margin: 20,
        borderTopRightRadius: 25,
        borderTopLeftRadius: 25,

    },
    box: {
        justifyContent: 'center',
        display: 'flex',
        flexDirection: 'column',
        flexWrap: 'wrap',
        alignItems: 'center',
        alignContent: 'center',
        width: '100%',
        borderTopLeftRadius: 40,
        borderTopRightRadius: 40,
        padding: 0,
    },
    row: {
        flexDirection: 'row',
    },
    column: {
        justifyContent: 'center',
        display: 'flex',
        flexDirection: 'column',
        flexWrap: 'wrap',
        alignItems: 'center',
        alignContent: 'center',
    },
    button: {
        justifyContent: 'center',
        borderBottomWidth: 1,
        padding: 12,
        margin: 25,
        width: 250,
        borderRadius: 5,
        borderColor: COLORS.second,
        backgroundColor: COLORS.second,
    },
    text: {
        textAlign: 'center',
        color: 'white',
    },
    textWelcome: {
        textAlign: 'center',
        color: COLORS.main,
        fontSize: 32,
        fontWeight: 'bold',
        margin: 10,
        marginBottom: 20,
    },
    textWelcome2: {
        textAlign: 'center',
        color: COLORS.main,
        fontSize: 12,
        fontWeight: 'bold',
    },
    textInput: {
        fontSize: 16,
        textAlign: 'left',
        color: COLORS.main,
        marginTop: 20,
    },
    input: {
        padding: 5,
        paddingLeft: 15,
        marginBottom: 10,
        borderBottomWidth: 2,
        borderRadius: 5,
        borderColor: COLORS.main,
        textAlign: 'left',
        color: COLORS.main,
    },
    logo: {
        justifyContent: 'center',
        alignItems: 'center',
        alignContent: 'center',
        width: 100,
        height: 75,
        resizeMode: 'contain',
        margin: 10,
    },
    logoContainer: {
        margin: 10,
        padding: 10,
    },
    register: {
        textAlign: 'center',
        color: COLORS.main,
    },
});
export default LoginScreen;