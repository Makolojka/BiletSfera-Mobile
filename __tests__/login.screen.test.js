import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from "../components/screens/LoginScreen";
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
test('renders correctly', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});

