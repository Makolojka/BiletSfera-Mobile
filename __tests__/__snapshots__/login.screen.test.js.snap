// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../components/screens/img/bg_1.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderTopLeftRadius": 25,
            "borderTopRightRadius": 25,
            "display": "flex",
            "flexDirection": "column",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "margin": 20,
            "padding": 10,
          }
        }
      >
        <View
          style={
            {
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../components/screens/img/logoWhite.png",
                }
              }
              style={
                {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 75,
                  "justifyContent": "center",
                  "margin": 10,
                  "resizeMode": "contain",
                  "width": 100,
                }
              }
            />
            <View
              style={
                {
                  "alignContent": "center",
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "margin": 10,
                    "marginBottom": 20,
                    "textAlign": "center",
                  }
                }
              >
                Witaj w BiletSferze
              </Text>
              <Text
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                Zaloguj się, aby kontynuować.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignContent": "center",
              "alignItems": "center",
              "borderTopLeftRadius": 40,
              "borderTopRightRadius": 40,
              "display": "flex",
              "flexDirection": "column",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "padding": 0,
              "width": "100%",
            }
          }
        >
          <View>
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "marginTop": 20,
                  "textAlign": "left",
                }
              }
            >
              Login
            </Text>
            <TextInput
              keyboardType="default"
              onChangeText={[Function]}
              style={
                {
                  "borderBottomWidth": 2,
                  "borderColor": "white",
                  "borderRadius": 5,
                  "color": "white",
                  "marginBottom": 10,
                  "padding": 5,
                  "paddingLeft": 15,
                  "textAlign": "left",
                }
              }
              value=""
            />
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "marginTop": 20,
                  "textAlign": "left",
                }
              }
            >
              Hasło
            </Text>
            <TextInput
              keyboardType="default"
              onChangeText={[Function]}
              secureTextEntry={true}
              style={
                {
                  "borderBottomWidth": 2,
                  "borderColor": "white",
                  "borderRadius": 5,
                  "color": "white",
                  "marginBottom": 10,
                  "padding": 5,
                  "paddingLeft": 15,
                  "textAlign": "left",
                }
              }
              value=""
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#6B3FA0",
                  "borderBottomWidth": 1,
                  "borderColor": "#6B3FA0",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "margin": 25,
                  "opacity": 1,
                  "padding": 12,
                  "width": 250,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "textAlign": "center",
                  }
                }
              >
                Zaloguj się
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateY": -48,
          },
        ],
        "zIndex": 99,
      }
    }
  />
</View>
`;
